src/ccalltest.c

```diff
@@ -26,7 +27,7 @@ int __declspec(noinline)
 #else
 int __attribute__((noinline))
 #endif
-testUcharX(unsigned char x) {
+DLLEXPORT testUcharX(unsigned char x) {
```

Comment:
Actually, the reason I added these `DLLEXPORT` things is because on Linux, with GCC, the symbols weren't accessible.  `nm` showed them as `'t'` instead of `'T'` which apparently means "local", but I'm honestly a little fuzzy as to what exactly that means.  It's possible that what Tony just said is the culprit; something else was getting exported which caused everything else to be non-visible, but at least things are more explicit this way.